pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -1950,91 +1950,3 @@ func TestChangedCompliantUsername(t *testing.T) {
 	// the CompliantUsername and MUR name should now match
 	require.Equal(t, userSignup.Status.CompliantUsername, mur.Name)
 }
-
-func TestMigrateMur(t *testing.T) {
-	// given
-	userSignup := &v1alpha1.UserSignup{
-		ObjectMeta: newObjectMeta("foo", "foo@redhat.com"),
-		Spec: v1alpha1.UserSignupSpec{
-			Username:      "foo@redhat.com",
-			Approved:      true,
-			TargetCluster: "east",
-		},
-	}
-	mur := newMasterUserRecord(basicNSTemplateTier, "foo", operatorNamespace, "east", "foo")
-	// override values to meet the test requirements.
-	mur.Spec.UserAccounts[0].Spec.NSTemplateSet.Namespaces = []v1alpha1.NSTemplateSetNamespace{
-		{
-			Type:        "dev",
-			Revision:    "123abc1",
-			TemplateRef: "",
-		},
-		{
-			Type:        "stage",
-			Revision:    "123abc2",
-			TemplateRef: "",
-		},
-		{
-			Type:        "extra",
-			Revision:    "123abc3",
-			TemplateRef: "",
-		},
-	}
-	mur.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources = &toolchainv1alpha1.NSTemplateSetClusterResources{
-		Revision:    "654321b",
-		TemplateRef: "",
-	}
-
-	expectedMur := mur.DeepCopy()
-	expectedMur.Generation = 1
-	expectedMur.ResourceVersion = "1"
-	expectedMur.Spec.UserAccounts[0].Spec.NSTemplateSet.Namespaces = []v1alpha1.NSTemplateSetNamespace{
-		{
-			TemplateRef: "basic-dev-123abc1",
-		},
-		{
-			TemplateRef: "basic-stage-123abc2",
-		},
-		{
-			TemplateRef: "basic-extra-123abc3",
-		},
-	}
-	expectedMur.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources = &toolchainv1alpha1.NSTemplateSetClusterResources{
-		TemplateRef: "basic-clusterresources-654321b",
-	}
-
-	t.Run("add missing templateRef fields", func(t *testing.T) {
-		// given
-		r, req, _ := prepareReconcile(t, userSignup.Name, userSignup, basicNSTemplateTier, mur)
-
-		// when
-		_, err := r.Reconcile(req)
-
-		// then verify that the MUR exists and is complete
-		require.NoError(t, err)
-		murs := &v1alpha1.MasterUserRecordList{}
-		err = r.client.List(context.TODO(), murs)
-		require.NoError(t, err)
-		require.Len(t, murs.Items, 1)
-		assert.Equal(t, *expectedMur, murs.Items[0])
-	})
-
-	t.Run("fails while updating mur that is missing templateRef fields", func(t *testing.T) {
-		// given
-		r, req, fakeClient := prepareReconcile(t, userSignup.Name, userSignup, basicNSTemplateTier, mur)
-		fakeClient.MockUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
-			return fmt.Errorf("some error")
-		}
-
-		// when
-		_, err := r.Reconcile(req)
-
-		// then verify that the MUR exists and is complete
-		require.Error(t, err)
-		murs := &v1alpha1.MasterUserRecordList{}
-		err = r.client.List(context.TODO(), murs)
-		require.NoError(t, err)
-		require.Len(t, murs.Items, 1)
-		assert.NotEqual(t, *expectedMur, murs.Items[0])
-	})
```

Comment:
I know that you removed the migration of `TemplateRef` part, but you could verify (also the error case) situation when either `NSLimit` or `TierName` is missing.

Reply:
@MatousJobanek i've added a test to make sure that the migration of nslimit and tiername happens here: 068aed7c2f3084cd0a0be2c71ca8837e39fd5b3a - is this good?